### PR TITLE
Remove jQuery lib and update code according to new JS checkout form function

### PR DESCRIPTION
**Pull request by guzzilar**
**1. Objective reason**

Some users could't use Omise's checkout form because their theme/template doesn't include/import `jQuery` library to the project at the initial step that our plugin required. 

So, this PR's gonna provide another way to detach the jQuery lib out from the project and implement pure JS function to handle checkout form instead.

**2. Description of change**

- Implement pure JS to handle checkout form
- Remove jQuery & update code according to new checkout form handler functions

**3. Users affected by the change**

`-`

**4. Impact of the change**

`-`

**5. Priority of change**

Normal

**6. Alternate solution**

`-`

**7. Tested on**
- Magento (community version) 1.9.2.4

checkout form
screen shot 2559-08-09 at 4 28 53 pm

checkout form with validation failed
screen shot 2559-08-09 at 4 28 08 pm

checkout form with `checkout.setLoadWaiting('payment');`
screen shot 2559-08-09 at 4 28 56 pm

checkout form with Omise's error (this one I bypassed a card's number value to Omise to let them throw an error, so from the image 👇 usually, it'll not pass to Omise server because validator gonna throw an validate error instead.)
screen shot 2559-08-09 at 4 36 12 pm

### CODE REVIEW

File: src/app/design/frontend/base/default/template/payment/form/omisecc.phtml [hunk 1]
```diff
@@ -1,77 +1,119 @@
...
+
+                    script.onload = script.onreadystatechange = null;
+                    head.removeChild(script);
+                };
+            };
+
+            head.appendChild(script);
+        };
+
+        getScript('https://ajax.googleapis.com/ajax/libs/jquery/1.12.4/jquery.min.js', function() {
```
Comment (nimid) [inline src/app/design/frontend/base/default/template/payment/form/omisecc.phtml]: Is a dependency, jQuery, required for using Omise.js?

Comment (guzzilar) [inline src/app/design/frontend/base/default/template/payment/form/omisecc.phtml]: hmm, let me check
thanks

File: src/app/design/frontend/base/default/template/payment/form/omisecc.phtml [hunk 2]
```diff
@@ -1,77 +1,119 @@
...
+    } else {
+        $nj = jQuery.noConflict();
+        hookPaymentSave($nj);
+    }
+
+    function hookPaymentSave(jQuery) {
+        Payment.prototype.save = Payment.prototype.save.wrap(function(superMethod) {
+            var paymentMethod = jQuery("#co-payment-form").find("input[name='payment[method]']:checked").val();
+
+            if (typeof paymentMethod == "undefined") {
```
Comment (nimid) [inline src/app/design/frontend/base/default/template/payment/form/omisecc.phtml]: Is it better to reduce this nested conditions and the long condition scope?

For an example:
```javascript
if (typeof paymentMethod == "undefined") {
    alert('Please select one method');
    return false;
}

if (paymentMethod != "omise_gateway") {
    return superMethod();
}

var omiseValidation = new Validation('co-payment-form');

// If validate pass
if (omiseValidation.validate()) {
...
```

File: src/app/design/frontend/base/default/template/payment/form/omisecc.phtml (hunk 1 above)
Comment (guzzilar) [inline src/app/design/frontend/base/default/template/payment/form/omisecc.phtml]: ok, I'll try to rewrite to remove those jQuery stuff,
I think it'd be better to not rely on 3rd-party lib. 👍

File: src/app/design/frontend/base/default/template/payment/form/omisecc.phtml (hunk 2 above)
Comment (guzzilar) [inline src/app/design/frontend/base/default/template/payment/form/omisecc.phtml]: @nimid good advice, I also like the way to move `else` condition to the top of the code block.
```js
if (typeof paymentMethod == "undefined") {
    alert('Please select one method');
    return false;
}

if (paymentMethod != "omise_gateway") {
    return superMethod();
}

var omiseValidation = new Validation('co-payment-form');

// If validate pass
if (omiseValidation.validate()) {
  // Take care only success case.
}
```

👍 

will fix it

**Comment by guzzilar** [comment]
FYI: changed PR's title from 
`Include jQuery to the checkout form in case that template doesn't support jQuery`
👇 
`Remove jQuery lib and update code according to new JS checkout form function`

also, updating PR's desc.

**Comment by guzzilar** [comment]
done, @nimid please review

thanks

File: src/app/design/frontend/base/default/template/payment/form/omisecc.phtml [hunk 3]
```diff
@@ -1,77 +1,93 @@
...
+        var validator = new Validation("co-payment-form");
+
+        if (validator.validate()) {
+            checkout.setLoadWaiting('payment');
+
+            getScript("https://cdn.omise.co/omise.min.js.gz", function() {
+                // Set Omise key
+                Omise.setPublicKey("<?php echo $this->getOmiseKeys('public_key'); ?>");
+
+                var form                  = document.getElementById("payment_form_omise_gateway"),
```
Comment (nimid) [inline src/app/design/frontend/base/default/template/payment/form/omisecc.phtml]: A variable, form, was defined here but I found no any place using it.
Could you please guide me where it is used?

Comment (nimid) [inline src/app/design/frontend/base/default/template/payment/form/omisecc.phtml]: @guzzilar
A variable, form, was defined here but I found no any place using it.
Could you please guide me where it is used?

Comment (guzzilar) [inline src/app/design/frontend/base/default/template/payment/form/omisecc.phtml]: it used to use for previous implementation.
Right now is no where, will remove it 👍

**Comment by guzzilar** [comment]
done & tested on 1.9.2.4

File: src/app/design/frontend/base/default/template/payment/form/omisecc.phtml [hunk 4]
```diff
@@ -117,7 +132,7 @@
             </select>
         </div>
 
-        <input data-omise="omise_token" type="hidden" name="payment[omise_token]">
+        <input id="<?php echo $_code ?>_token" data-omise="omise_token" type="hidden" name="payment[omise_token]">
```
Comment (nimid) [inline src/app/design/frontend/base/default/template/payment/form/omisecc.phtml]: Now, the attribute, id, was used to indicate and manipulate so the attribute, data-omise, can be removed from all elements.

Comment (guzzilar) [inline src/app/design/frontend/base/default/template/payment/form/omisecc.phtml]: @nimid at first I don't want to touch any html elements much (just solve jQuery issue).
But, agree, will remove it. 👍

**Comment by guzzilar** [comment]
@nimid for your comment about `data-omimse` attributes,
done here: https://github.com/omise/omise-magento/pull/24/commits/d4e1a72a5ef343576beb7d989b34eec8287cdadd

I also did code styling with checkout form a bit here https://github.com/omise/omise-magento/pull/24/commits/d83b1e4af55fe875a75388269a30cfb143e0ea49


thanks

**Comment by guzzilar** [comment]
tested on 1.9.2.4

checkout form
screen shot 2559-08-09 at 4 28 53 pm

checkout form with validation failed
screen shot 2559-08-09 at 4 28 08 pm

checkout form with `checkout.setLoadWaiting('payment');`
screen shot 2559-08-09 at 4 28 56 pm

checkout form with Omise's error (this one I bypassed a card's number value to Omise to let them throw an error, so from the image 👇 usually, it'll not pass to Omise server because validator gonna throw an validate error instead.)
screen shot 2559-08-09 at 4 36 12 pm

**Comment by guzzilar** [comment]
FYI, updated PR's desc

**Comment by nimid** [comment]
I have test with web browsers, Safari 9.1.2, Opera 39.0, Firefox 47.0.1 and Google Chrome 52.0.
The result is passed. It work well with the theme that contains jQuery and not contains jQuery.

Below are the screenshots show the testing with the theme that has no jQuery.

1. Fill the information to the checkout form
screenshot-192 168 99 100 8081 2016-08-09 18-04-41

2. Processing checkout
screenshot-192 168 99 100 8081 2016-08-09 18-04-57

3. Checkout is completed
screenshot-192 168 99 100 8081 2016-08-09 18-05-14
